### PR TITLE
Cortx 33515 testcases

### PR DESCRIPTION
**Pull request by pragamj**
# Problem Statement
- TA TEST-42910[Data block corruption in healthy mode and detection with M0cp and M0cat in Degraded Mode]

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True 
[collections_log.txt](https://github.com/Seagate/cortx-test/files/9348033/collections_log.txt)
--collect-only`)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by pranav-sahasrabudhe — APPROVED** [review]

File: comptests/motr/test_motr_di.py [hunk 1]
```diff
@@ -183,3 +188,59 @@ def test_m0cp_m0cat_block_corruption_unaligned(self):
...
+        """
+        Corrupt data block using m0cp and reading from object with m0cat should error.
+        -s 4096 -c 10 -o 1048583 /root/infile -L 1
+        -s 4096 -c 1 -o 1048583 /root/myfile -L 1 -u -O 0
+        -o 1048583 -s 4096 -c 10 -L 1 /root/dest_myfile
+        """
+        count_list = ['8']
+        bsize_list = ['4K']
+        layout_ids = ['1']
+        proc_restart_delay = 9999
```
Comment (priyankaborawake) [inline comptests/motr/test_motr_di.py:205]: Can this be moved to config file ?

File: comptests/motr/test_motr_di.py [hunk 2]
```diff
@@ -183,3 +188,59 @@ def test_m0cp_m0cat_block_corruption_unaligned(self):
...
+                            str(self.system_random.randint(1, 1024 * 1024))
+                for b_size, cnt_c, layout in zip(bsize_list, count_list, layout_ids):
+                    self.motr_obj.dd_cmd(
+                        b_size, cnt_c, infile, node)
+                    object_id_list.append(object_id)
+                    self.motr_obj.cp_cmd(
+                        b_size, cnt_c, object_id,
+                        layout, infile, node, client_num)
+        # Degrade the setup by killing the m0d process
+        data_pod_list = self.motr_obj.master_node_list[0].select_random_pod_container(
```
Comment (priyankaborawake) [inline comptests/motr/test_motr_di.py:225]: Any specific purpose of adding -001?

Comment (pragamj) [reply comptests/motr/test_motr_di.py:310]: yes to go to motr-io-001 container

File: comptests/motr/test_motr_di.py [hunk 3]
```diff
@@ -266,3 +273,59 @@ def test_data_block_corruption_one_by_one(self):
             # Read the data using m0cp utility
             self.m0cat_md5sum_m0unlink(bsize_list, count_list, layout_ids, object_list,
                                        client_num=client_num)
+
+    @pytest.mark.skip(reason="Test incomplete without teardown")
+    @pytest.mark.tags("TEST-42910")
+    @pytest.mark.motr_di
+    def test_m0cp_block_corruption_m0cat_degraded_mode(self):
```
Comment (priyankaborawake) [inline comptests/motr/test_motr_di.py:280]: As discussed, Edit the deployment file using below method in the setup method. 
edit_deployments_for_delay

File: comptests/motr/test_motr_di.py [hunk 4]
```diff
@@ -266,3 +273,59 @@ def test_data_block_corruption_one_by_one(self):
...
+                    self.motr_obj.dd_cmd(
+                        b_size, cnt_c, infile, node)
+                    object_id_list.append(object_id)
+                    self.motr_obj.cp_cmd(
+                        b_size, cnt_c, object_id,
+                        layout, infile, node, client_num)
+        # Degrade the setup by killing the m0d process
+        data_pod_list = self.motr_obj.master_node_list[0].select_random_pod_container(
+            POD_NAME_PREFIX, f"{MOTR_CONTAINER_PREFIX}-001")
+        pod_name, container = secrets.choice(data_pod_list)
```
Comment (priyankaborawake) [inline comptests/motr/test_motr_di.py:312]: We are already selecting random pod and container, no need for secrets.choice

File: comptests/motr/test_motr_di.py [hunk 5]
```diff
@@ -266,3 +273,59 @@ def test_data_block_corruption_one_by_one(self):
...
+            self.dtm_obj.set_proc_restart_duration(
+                self.motr_obj.master_node_list[0], pod_name, container, 0)
+            assert False
+        # Read the data using m0cat in degraded mode
+        for client_num in range(motr_client_num):
+            for node in node_pod_dict:
+                for b_size, cnt_c, layout, object_id in zip(bsize_list, count_list, layout_ids,
+                                                            object_id_list):
+                    self.motr_obj.cat_cmd(b_size, cnt_c, object_id, layout,
+                                          outfile, node, client_num)
```
Comment (priyankaborawake) [inline comptests/motr/test_motr_di.py:331]: Add a new line here.

**Review by priyankaborawake — CHANGES_REQUESTED** [review]

File: comptests/motr/test_motr_di.py (hunk 4 above)
Comment (pragamj) [reply comptests/motr/test_motr_di.py:312]: done

File: comptests/motr/test_motr_di.py (hunk 5 above)
Comment (pragamj) [reply comptests/motr/test_motr_di.py:331]: done

File: comptests/motr/test_motr_di.py (hunk 3 above)
Comment (pragamj) [reply comptests/motr/test_motr_di.py:280]: need to change as per v0.09.0

Comment (pragamj) [reply comptests/motr/test_motr_di.py:279]: will pick in RBF task

**Review by priyankaborawake — APPROVED** [review]

**Review by ketanarlulkar — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**